cdap-api/src/main/java/co/cask/cdap/api/schedule/Schedules.java

```diff
@@ -64,4 +64,87 @@ private Schedules() {
   public enum Source {
     STREAM
   }
+
+  /**
+   * Get a builder used to create a schedule.
+   *
+   * @param name the name of the schedule to build
+   * @return a builder used to create a schedule
+   */
+  public static Builder builder(String name) {
+    return new Builder(name);
+  }
+
+  /**
+   * Builder used to create a schedule.
+   */
+  public static class Builder {
+    private final String name;
+    private String description;
+    private Integer concurrentProgramRunsThreshold;
+
+    private Builder(String name) {
+      this.name = name;
+      this.description = "";
+    }
+
+    /**
+     * Set the schedule description.
+     *
+     * @param description the schedule description
+     * @return the Builder
+     */
+    public Builder setDescription(String description) {
+      this.description = description;
+      return this;
+    }
+
+    /**
+     * Skip running the scheduled program if the number of concurrent runs of the program exceeds the specified
+     * threshold. Prior to running the program, the scheduler will check how many runs of the program are currently
+     * in the RUNNING state.
+     * If that number is greater than the specified threshold, the scheduler will not run the program.
+     * This number includes runs of the program that were started outside of this schedule,
+     * either manually or by another schedule. For example, if the threshold is set to 0, the scheduler will
+     * only run the program if there are no other runs of the program currently in the RUNNING state.
+     *
```

Comment:
So if I understand correctly, what will happen is:
1. Time schedule triggers
2. Scheduler checks how many runs are currently in RUNNING state. If that exceeds the threshold: skip this run
3. Start the new run
4. New run goes into STARTING state
5. New run goes into RUNNING state
 
I think there can be a race condition: During step 2, some runs may be submitted but not in RUNNING state yet. Perhaps STARTING. I think step 2 should at least count STARTING runs in addition to RUNNING ones. Even with that, another run could be between steps 3 and 4 (submitted but not STARTING yet). Or is that not possible?